### ISSUE TITLE
wezterm size changes when opening other app

### ISSUE THREAD

**Issue by Naadiyaar** [wez/wezterm#4568]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

GNOME Mutter 44.6

### WezTerm version

20231107-082518-f0e3eecb

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

When full screen, and opening(focusing) other programs, wezterm loses position and size.
[Screencast from 2023-11-12 17-46-54.webm](https://github.com/wez/wezterm/assets/106643149/3aef772e-6fb2-409d-8962-e1fde0dc060e)

### To Reproduce

open wezterm
make it full-screen or half screen
focus on other program

### Configuration

THIS IS FULL CONFIG
```lua
local cfg = {
	window_close_confirmation = "NeverPrompt",
	check_for_updates = false,

	-- Avoid unexpected config breakage
	automatically_reload_config = false,

	exit_behavior = "Close",

	-- Disable all noises
	audible_bell = "Disabled",

	-- remove terminal top bar
	window_decorations = "NONE",

	-- setup color theme
	color_scheme = "Solarized Light (Gogh)",
}
cfg.hide_tab_bar_if_only_one_tab = true

cfg.window_padding = {
	left = 3,
	right = 3,
	top = 3,
	bottom = 3,
}

cfg.inactive_pane_hsb = {
	saturation = 0.9,
	brightness = 0.6,
}

-- set cursor shape to a Blinking Bar
cfg.default_cursor_style = "BlinkingBar"
cfg.animation_fps = 1
cfg.cursor_blink_ease_in = "Constant"
cfg.cursor_blink_ease_out = "Constant"

return cfg
```
plus some things about font

### Expected Behavior

Secondary program gain focus while wezterm keeps its shape

### Anything else?

It's not happening with no configuration

**Comment by Naadiyaar** [comment]
If it helps to solve the problem, also moving the terminal window to another workspace fixes the problem.
Like 
opening the terminal
make it full screen
move it to another workspace

**Comment by Naadiyaar** [comment]
issue is solved by removing `window_decorations = "NONE"` 
any idea what's causing this?

**Comment by jokeyrhyme** [comment]
I'm on Arch Linux, GNOME 45.1 (wayland), wezterm 20230712-072601-f4abf8fd and I can't reproduce this, so I wonder if this has been fixed on the GNOME side? Are you able to boot GNOME OS or some other live USB with a newer GNOME to check?

**Comment by Naadiyaar** [comment]
currently i'm using a fresh install Fedora 39 with GNOME 45.1 and this behavior is easily reproducible by adding `window_decorations = "NONE"` to config. 
so no it's not fixed here

**Comment by Naadiyaar** [comment]
(potentially) more discussion here
https://discussion.fedoraproject.org/t/wezterm-miss-behavieor-when-setting-window-decoration/97115

**Comment by aliaksandr-trush** [comment]
Should be fixed by #5897

**Comment by wez** [comment]
Please try the latest nightly build
